packages/x-date-pickers/src/internals/components/PickersViewLayout/PickersViewLayout.types.ts

```diff
@@ -43,7 +43,7 @@ interface PickersViewLayoutSlotsComponentsProps
   /**
    * Props passed down to the toolbar component.
    */
-  toolbar?: ExportedBaseToolbarProps;
+  toolbar?: ExportedBaseToolbarProps & Record<string, any>;
```

Comment:
Otherwise, the typing in the range hooks complains that we try to pass additional props
This interface is private only so I don't think it's an issue to add the record

Reply:
Technically this is actually public. It's exposed via `themeAugmentation` and would allow code like this:
```ts
MuiPickersViewLayout: {
  defaultProps: {
    componentsProps: {
      toolbar: {
        // should complain that prop does not exist
        something: 'something'
      }
    }
  },
},
```

Honestly, I would probably be in favour maybe casting whatever is passed in or coming up with a different solution  🤷 🙈

Reply:
You are right, I replace it with type casting.